### PR TITLE
Better tests

### PR DESCRIPTION
**Pull request by yanefingon**
The last one focuses on the addition of a more robust and unpredictable test (as those used in an original work were not capable of detecting the bug which is a shame).

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -14,6 +14,10 @@ pub trait StalinFind<T> {
     if self.is_empty() {
       None
     } else {
+      self.sort() // so that the "binarity" of the search is finally useful
```
Comment (Miezhiko) [inline src/lib.rs]: no, this increases complexity a lot, I think it's meaningless to do, you can use alternative find algos if array isn't sorted or not fully sorted. (tho if array partially sorted you may have luck that this algo will work fast as well).

File: src/vec.rs [hunk 2]
```diff
@@ -13,32 +13,65 @@ impl<T: PartialOrd> StalinFind<T> for Vec<T> {
...
-    if self[m] == i {
-      Some(m)
-    } else {
-      self.remove(m);
-      if m == 0 {
-        if self.is_empty() {
-          None
-        } else {
-          self.stalin(i, 0, self.len() - 1)
+    fn stalin(&mut self, i: T, l: usize, r: usize) -> Option<usize>
```
Comment (Miezhiko) [inline src/vec.rs]: you are not following original code formatting, I'm using two spaces and open bracket on same line... =_=

File: src/vec.rs [hunk 3]
```diff
@@ -13,32 +13,65 @@ impl<T: PartialOrd> StalinFind<T> for Vec<T> {
...
+    let mut rng = rand::thread_rng();
+    let sought = rng.gen_range(0, n);
+    let amount = rng.gen_range(0, n);
+    let range = Uniform::new(0, n);
+    let mut vector: Vec<u64> = (0..amount).map(|_| rng.sample(&range)).collect();
+
+    // check if the value is within the array
+    let i = vector.iter().position(|x| x == &sought);
+    if i.is_some() {
+        // IM NOT SURE HERE AS WELL THE LANGUAGE IS STUPID EVERYTHING IS STUPID I HATE IT
```
Comment (Miezhiko) [inline src/vec.rs]: is this inspired by my C++ code comments? ^_^

**Comment by Miezhiko** [comment]
@yanefingon MERGFED, thanks

File: src/lib.rs (hunk 1 above)
Comment (yanefingon) [reply src/lib.rs]: > you may have luck that this algo will work fast as well

sure. as one might have with the bogglesort

File: src/vec.rs (hunk 2 above)
Comment (yanefingon) [reply src/vec.rs]: sure

File: src/vec.rs (hunk 3 above)
Comment (yanefingon) [reply src/vec.rs]: gotta patent the ARG styling pattern egh?

**Pull request merged**